jvm/src/main/java/com/microsoft/bond/protocol/CompactBinaryProtocolWriter.java

```diff
@@ -0,0 +1,383 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+import com.microsoft.bond.Metadata;
+import com.microsoft.bond.ProtocolType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Stack;
+
+// TODO: add Compact Binary protocol format comment (either here and/or in the deserializer and/or in package.html)
```

Comment:
Personally, I don't think the description of the protocol belongs anywhere language-specific. I know we have ASCII art in the equivalent places in other languages and I'd like to break with that tradition.